packages/labs/eleventy-plugin-lit/src/index.ts

```diff
@@ -50,83 +133,117 @@ module.exports = {
+        'lit/directives/unsafe-html.js',
+        renderModulePath
+      )
+    ).module.namespace.unsafeHTML as typeof contextifiedUnsafeHTML;
+  });
+
+  eleventyConfig.addTransform(
+    'render-lit',
+    async (content: string, outputPath: string) => {
+      if (!outputPath.endsWith('.html')) {
+        return content;
+      }
+
+      // TODO(aomarks) Maybe we should provide a `renderUnsafeHtml` function
+      // directly from SSR which does this.
+      const iterator = contextifiedRender(contextifiedUnsafeHTML(content));
+      const concatenated = iterableToString(iterator);
+      // Lit SSR includes comment markers to track the outer template from
+      // the template we've generated here, but it's not possible for this
+      // outer template to be hydrated, so they serve no purpose.
+
+      // TODO(aomarks) Maybe we should provide an option to SSR option to skip
+      // outer markers (though note there are 2 layers of markers due to the
+      // use of the unsafeHTML directive).
+      const outerMarkersTrimmed = concatenated
+        .replace(/^((<!--[^<>]*-->)|(<\?>)|\s)+/, '')
+        .replace(/((<!--[^<>]*-->)|(<\?>)|\s)+$/, '');
+      return outerMarkersTrimmed;
     }
+  );
+}
 
+module.exports = {
+  configFunction: function (
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    eleventyConfig: any,
+    {componentModules, mode = 'worker'}: LitPluginOptions = {}
+  ) {
     if (componentModules === undefined || componentModules.length === 0) {
       // If there are no component modules, we could never have anything to
       // render.
       return;
     }
 
-    const renderModulePath = path.join(process.cwd(), 'arbitrary.js');
     const resolvedComponentModules = componentModules.map((module) =>
       path.resolve(process.cwd(), module)
     );
 
-    let contextifiedRender: (value: unknown) => IterableIterator<string>;
-    let contextifiedUnsafeHTML: (value: string) => unknown;
-
-    // Create a fresh context before each build, so that our module cache resets
-    // on every --watch mode build.
-
-    // TODO(aomarks) For better performance, we could re-use contexts between
-    // build, but selectively invalidate its cache so that only the user's
-    // modules are reloaded.
-    eleventyConfig.on('eleventy.before', async () => {
-      const {getWindow} = await import('@lit-labs/ssr/lib/dom-shim.js');
-      const {ModuleLoader} = await import('@lit-labs/ssr/lib/module-loader.js');
-      const window = getWindow({includeJSBuiltIns: true});
-      const loader = new ModuleLoader({global: window});
-      // TODO(aomarks) Replace with concurrent Promise.all version once
-      // https://github.com/lit/lit/issues/2549 has been addressed.
-      for (const module of resolvedComponentModules) {
-        await loader.importModule(module, renderModulePath);
+    switch (mode) {
+      case 'worker': {
+        return configureWorker(eleventyConfig, resolvedComponentModules);
```

Comment:
I think I would remove the `return`s here and below, since nothing is returned.

Reply:
I was debating between doing a `return` here versus adding a `break;` after each of these. Now I agree that semantically `return` makes less sense as nothing's returned.